### PR TITLE
All (visible) modal views styled.

### PR DESCRIPTION
**Pull request by Markvanderuit**
Completely reworked style to be cleaner etc, blablabla design stuff.
Applied style to all modal views except ShotDecouplingModalView. Don't know where that one belongs yet.

Apologies for the excessive lines of code, but it's kind of one big feature.

### CODE REVIEW

**Comment by martijnjanssen** [comment]
Changes look good, but for the next time, please do not commit unused code yet. This code is not tested or used. If you start using it the next time, it does not show up in the commit log. We do not know what it will precisely do.

And please, make smaller pull requests, this is way too much. I get that this is mostly styling, but there is just too much going on. Maybe styling 2-3 windows per pull request is okay. We also like to have these awesome changes as soon as possible!

**Comment by Markvanderuit** [comment]
Yeah, that's definitely not happening again. Ever. Third time will have me buying cake.

File: src/main/java/gui/misc/ResizeHelper.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        stage.getScene().addEventHandler(MouseEvent.MOUSE_PRESSED, getOnMousePressedHandler());
+        stage.getScene().addEventHandler(MouseEvent.MOUSE_EXITED, getOnMouseExitHandler());
+    }
+
+    /**
+     * Updates necessary variables used by all event handlers.
+     * @param e the event whose data we use.
+     */
+    private void updateVariables(MouseEvent e) {
+
```
Comment (geenen124) [inline src/main/java/gui/misc/ResizeHelper.java]: Unnecessary line?

Comment (Markvanderuit) [inline src/main/java/gui/misc/ResizeHelper.java]: Fixed.

File: src/main/java/gui/misc/TransitionHelper.java [hunk 2]
```diff
@@ -25,20 +27,36 @@
 
     @Getter
     private Node node;
+    @Getter
+    private ArrayList<EventHandler> addedHandlers;
```
Comment (geenen124) [inline src/main/java/gui/misc/TransitionHelper.java]: Use the interface that you want to use, as opposed to the implementing class. `List` in this case.

Comment (Markvanderuit) [inline src/main/java/gui/misc/TransitionHelper.java]: `private List<EventHandler> addedHandlers;`
and
`this.addedHandlers = new ArrayList<EventHandler>();`
So fixed.

File: src/main/java/gui/modal/DirectorShotCreationModalView.java [hunk 3]
```diff
@@ -36,36 +36,37 @@
...
 
+
+    // three main colors used throughout window. Experiment a little!
+    private Color mainColor = Color.rgb(255, 172, 70); // main bright color
+    private Color secondaryColor = Color.rgb(255, 140, 0); // darker color
+    private Color tertiaryColor = Color.rgb(255, 235, 190); // lighter color
+
+    // variables for spacing
+    private int topAreaHeight = 70;
+    private int bottomAreaHeight = 60;
```
Comment (geenen124) [inline src/main/java/gui/modal/DirectorShotCreationModalView.java:49]: If these aren't re-assigned, make them `static final` variables. Same goes for the rest of these private fields (Color, style strings etc.)

File: src/main/java/gui/root/RootPane.java [hunk 4]
```diff
@@ -1,10 +1,13 @@
 package gui.root;
 
+
+
```
Comment (geenen124) [inline src/main/java/gui/root/RootPane.java]: Unnecessary lines?

File: src/main/java/gui/modal/DirectorShotCreationModalView.java (hunk 3 above)
Comment (Markvanderuit) [inline src/main/java/gui/modal/DirectorShotCreationModalView.java:49]: Plan right now is to push most of these out of these classes somehow, so we can set application colors globally. But yeah, for now I'll make them static final.

File: src/main/java/gui/root/RootPane.java (hunk 4 above)
Comment (Markvanderuit) [inline src/main/java/gui/root/RootPane.java]: Fixed.

File: src/main/java/gui/modal/DirectorShotCreationModalView.java (hunk 3 above)
Comment (Markvanderuit) [inline src/main/java/gui/modal/DirectorShotCreationModalView.java:49]: Same for the rest of the Modal views, so you don't have to comment on the same issue there.

**Comment by geenen124** [comment]
Keep the PR size smaller next time, but looks good!

**Comment by bart-de-jonge** [comment]
I know some team members pointed this out already, but I want to emphasize: make smaller PR's. We commented about this on a previous PR as well

**Pull request merged**